### PR TITLE
add support for new wrap options in tests and heredoc support for sharness

### PR DESCRIPTION
**Pull request by trws**
This aggregates a couple of docker-run-checks, flux-sharness and sharness updates.

### CODE REVIEW

**Comment by trws** [comment]
This has turned into a rabbit hole, and it turns out the feature I thought the new sharness had, it doesn't. So I'm building it and probably just going to revert the sharness update part of this.  It's a ton of strange issues and possible churn for, in the end, not much.  Updates to come.

**Comment by grondo** [comment]
This is nice. I'd definitely suggest proposing it upstream.

**Comment by trws** [comment]
@grondo anything more you'd like to see here?

**Review by grondo — APPROVED** [review]
Sorry, this one just fell off my radar. LGTM!

File: t/tools/aggregate-results.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (trws) [reply t/tools/aggregate-results.sh]: Ah, this is a mistake. Sharness 1.2 requires it here, and some other helpers, I missed removing it. Thanks for finding that!

**Pull request merged**